### ISSUE TITLE
Support generating lines of CSV as an iterator

### ISSUE THREAD

**Issue by iamed2** [JuliaData/CSV.jl#608]
It came up in https://github.com/invenia/LibPQ.jl/pull/172 that it would be useful to have a version of `CSV.write` that generates rows iteratively. So this new `CSV.iterwrite` function would work like:

```julia
collect(CSV.iterwrite(table)) == readlines(seekstart(CSV.write(IOBuffer(), table)))
```

This seems pretty difficult to shoehorn in to the performance-optimized `CSV.write` code though.

**Comment by quinnj** [comment]
Initial PR to support this: https://github.com/JuliaData/CSV.jl/pull/611

**Comment by caseykneale** [comment]
Oh wow you beat me too it Eric - I have a similar but related issue. I ran into this problem at work in a bad way.

I had a 100k CSV with ~2k columns. The file could not be saved with CSV.write - it crashed Julia over and over despite having plenty of memory.. 

I was able to use eachrow() and write using delimitedfiles.jl in <1 min

Not the biggest invoncenience, but there are edge cases which freeze CSV.jl completely and requires me to reload a large dataset and try again.
